contracts/lib.rs

```diff
@@ -10,11 +10,102 @@ mod open_payroll {
     use ink::storage::traits::StorageLayout;
     use ink::storage::Mapping;
 
+    // Define the types that will be used in the contract
     type Multiplier = u128;
     type MultiplierId = u32;
+    // Establish the maximum number of beneficiaries and multipliers that can be added to the contract
     const MAX_BENEFICIARIES: usize = 100;
     const MAX_MULTIPLIERS: usize = 10;
 
+    /// Emitted when a beneficiary claims their payment
+    #[ink(event)]
+    pub struct Claimed {
+        #[ink(topic)]
+        account_id: AccountId,
+        amount: Balance,
+        total_payment: Balance,
+        claiming_period_block: BlockNumber,
+    }
+
+    /// Emitted when a multiplier is deactivated
+    #[ink(event)]
+    pub struct MultiplierDeactivated {
+        #[ink(topic)]
+        multiplier_id: MultiplierId,
+        valid_until_block: BlockNumber,
+    }
+
+    /// Emitted when a multiplier is deleted
+    #[ink(event)]
+    pub struct MultiplierDeleted {
+        #[ink(topic)]
+        multiplier_id: MultiplierId,
+        valid_until_block: BlockNumber,
+    }
+
+    /// Emiited when the ownership of the contract is transferred
+    #[ink(event)]
+    pub struct OwnershipTransferred {
```

Comment:
Maybe rename to OwnershipProposed or something like this. The fields should be current_owner and proposed owner